test/test_protocolzoo_entanglement_consumer.jl

```diff
@@ -12,17 +12,19 @@ if isinteractive()
 end
 
 
+
 for n in 3:30
-    net = RegisterNet([Register(10) for j in 1:n])
+    regsize = 10
+    net = RegisterNet([Register(regsize) for j in 1:n])
     sim = get_time_tracker(net)
 
     for e in edges(net)
-        eprot = EntanglerProt(sim, net, e.src, e.dst; rounds=-1, randomize=true, margin=5, hardmargin=3)
+        eprot = EntanglerProt(sim, net, e.src, e.dst; rounds=-1, randomize=true)
```

Comment:
It was introduced to avoid deadlocks and since we have a `DecoherenceProtocol` now, I thought not having extra parameters in the Entangler would be best.

Reply:
Plus, it simplifies the code to some degree

Reply:
I would say these are orthogonal concerns. Protocols should be useful on their own, and there are situations in which this particular protocol does not need a decoherence tracking protocol (e.g. the chain example). Actually, this particular change probably breaks these other examples and breaks tests.

Also, this would be a change requiring a breaking release. I want us to be more careful with API stability.